### PR TITLE
feat: add Stripe ExpressCheckoutElement

### PR DESCRIPTION
**Pull request by jamesgorrie**
- Adds the [`ExpressCheckoutElement`](https://docs.stripe.com/elements/express-checkout-element)
- Integrates with our `/create` endpoint
- Adds a switch so we can start with it off, or turn it off any time (There's quite a lot of test files touched to allow for this)

The switch requires people to update their `~/.gu/support-admin-console/switches_v2.json` to [match this JSON](https://gist.github.com/jamesgorrie/65732cb1fe84f791307ab002d6816683#file-switches_v2_express_checkout-json).

https://github.com/user-attachments/assets/d93221f2-56ae-4fd6-a374-b15f53456fd8

### Apple pay
Screenshot 2024-07-25 at 09 54 58

### Apple and Google Pay mobile
Screenshot 2024-07-25 at 09 53 23

### Apple and Google Pay desktop
Screenshot 2024-07-25 at 09 53 02

### CODE REVIEW

File: support-frontend/assets/pages/[countryGroupId]/checkout.tsx [hunk 1]
```diff
@@ -698,6 +726,30 @@ function CheckoutComponent({ geoId, appConfig }: Props) {
...
+		if (
+			paymentMethod === 'StripeExpressCheckoutElement' &&
+			stripe &&
+			elements
+		) {
+			const { paymentMethod } = await stripe.createPaymentMethod({
+				elements,
+			});
+			if (!paymentMethod) {
+				return;
```
Comment (andrewHEguardian) [inline support-frontend/assets/pages/[countryGroupId]/checkout.tsx:738]: I'm slightly lost in the diff here, what does this return do?

Comment (jamesgorrie) [reply support-frontend/assets/pages/[countryGroupId]/checkout.tsx]: Nice catch, this should be error handling which I have just added

File: support-frontend/assets/__mocks__/settingsMock.ts [hunk 2]
```diff
@@ -6,13 +6,15 @@ export const mockSettings = {
...
 			stripePaymentRequestButton: 'On',
+			stripeExpressCheckout: 'On',
 			payPal: 'On',
 			amazonPay: 'On',
 		},
 		recurringPaymentMethods: {
 			stripe: 'On',
 			stripeApplePay: 'On',
 			stripePaymentRequestButton: 'On',
+			stripeExpressCheckout: 'On',
```
Comment (jamesgorrie) [inline support-frontend/assets/__mocks__/settingsMock.ts:17]: These are for jest.

File: support-frontend/assets/pages/[countryGroupId]/checkout.tsx [hunk 3]
```diff
@@ -901,6 +957,109 @@ function CheckoutComponent({ geoId, appConfig }: Props) {
 						>
 							<Box cssOverrides={shorterBoxMargin}>
 								<BoxContents>
+									{useStripeExpressCheckout && (
+										<>
+											<ExpressCheckoutElement
+												onReady={({ availablePaymentMethods }) => {
+													console.info(availablePaymentMethods);
```
Comment (jamesgorrie) [inline support-frontend/assets/pages/[countryGroupId]/checkout.tsx:964]: Remove this and add a note why we have this.

**Review by andrewHEguardian — APPROVED** [review]
Great!

**Pull request merged**